src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -1522,4 +1528,82 @@ public ChronicleQuestionnaire getQuestionnaire( UUID studyId, UUID questionnaire
+            UUID participantESID = Preconditions
+                    .checkNotNull( entitySetsApi.getEntitySetId( participantESName ),
+                            "participant entity set does not exist" );
+
+            ListMultimap<UUID, Map<UUID, Set<Object>>> entities = ArrayListMultimap.create();
+            ListMultimap<UUID, DataAssociation> associations = ArrayListMultimap.create();
+
+            OffsetDateTime dateTime = OffsetDateTime.now();
+
+            List<UUID> questionEntityKeyIds = new ArrayList<>( questionnaireResponses.keySet() );
+            for ( int i = 0; i < questionEntityKeyIds.size(); i++ ) {
+                UUID questionEntityKeyId = questionEntityKeyIds.get( i );
+
+                Map<UUID, Set<Object>> answerEntity = ImmutableMap.of(
+                        valuesPTID,
+                        questionnaireResponses.get( questionEntityKeyId ).get( VALUES_FQN ) );
+                entities.put( answersESID, answerEntity );
+
+                // 1. create participant -> respondsWith -> answer association
+                Map<UUID, Set<Object>> respondsWithEntity = ImmutableMap.of(
+                        dateTimePTID,
+                        ImmutableSet.of( dateTime )
+                );
+                associations.put( respondsWithESID, new DataAssociation(
+                        participantESID,
+                        java.util.Optional.empty(),
+                        java.util.Optional.of( participantEKID ),
+                        answersESID,
+                        java.util.Optional.of( i ),
+                        java.util.Optional.empty(),
+                        respondsWithEntity
+                ) );
+
+                // 2. create answer -> addresses -> question association
+                Map<UUID, Set<Object>> addressesEntity = ImmutableMap.of(
+                        completedDateTimePTID,
+                        ImmutableSet.of( dateTime )
+                );
+                associations.put( addressesESID, new DataAssociation(
+                        answersESID,
+                        java.util.Optional.of( i ),
+                        java.util.Optional.empty(),
+                        questionsESID,
+                        java.util.Optional.empty(),
+                        java.util.Optional.of( questionEntityKeyId ),
+                        addressesEntity
+                ) );
+            }
+            DataGraph dataGraph = new DataGraph( entities, associations );
+            dataApi.createEntityAndAssociationData( dataGraph );
+
+            logger.info( "submitted questionnaire: studyId = {}, participantId = {}", studyId, participantId );
+        } catch ( Exception e ) {
+            String errorMsg = "an error occurred while attempting to submit questionnaire";
+            logger.error( errorMsg, e );
+            throw new RuntimeException( errorMsg );
+        }
+        return true;
```

Comment:
What's the point of returning a boolean when the logic is to either return true or to throw? Is the boolean return value used for anything? It doesn't seem like it. Please change method signature.

```
public void submitQuestionnaire
```

Reply:
Makes sense!